packages/babel-plugin/src/visitors/stylex-create/index.js

```diff
@@ -113,6 +113,21 @@ export default function transformStyleXCreate(
       memberExpressions,
     });
 
+    // add injection that mark variables used for dynamic styles as `inherits: false`
+    const injectedInheritStyles: { [string]: InjectableStyle } = {};
+    if (fns != null) {
+      const dynamicFnsNames = Object.values(fns)
+        ?.map((entry) => Object.keys(entry[1]))
+        .flat();
+      dynamicFnsNames.forEach((fnsName) => {
+        injectedInheritStyles[fnsName] = {
+          priority: 0,
+          ltr: `@property ${fnsName} { inherits: false }`,
+          rtl: null,
+        };
+      });
+    }
+
     if (!confident) {
       throw path.buildCodeFrameError(messages.NON_STATIC_VALUE, SyntaxError);
     }
```

Comment:
Make sure you add this code *after* the `if` condition that checks `confident`. Otherwise you'll end up throwing errors because `fns` will be `null` or `undefined`.

```suggestion
    if (!confident) {
      throw path.buildCodeFrameError(messages.NON_STATIC_VALUE, SyntaxError);
    }

    // add injection that mark variables used for dynamic styles as `inherits: false`
    const injectedInheritStyles: { [string]: InjectableStyle } = {};
    if (fns != null) {
      const dynamicFnsNames = Object.values(fns)
        ?.map((entry) => Object.keys(entry[1]))
        .flat();
      dynamicFnsNames.forEach((fnsName) => {
        injectedInheritStyles[fnsName] = {
          priority: 0,
          ltr: `@property ${fnsName} { inherits: false }`,
          rtl: null,
        };
      });
    }
```

Reply:
Great call, thanks for fixing it!